### PR TITLE
Implemented data(name, value)

### LINKED ISSUE #4: Create a `.data()` method
> This is an issue ready for newcomers and people who want to learn how to contribute to an open source project. Read more about this type of issues in [Up For Grabs](http://up-for-grabs.net)

Create a function similar to `attr()` that is just syntax-sugar for the `data-` attributes. This *does not* behave like jquery's (in jquery's there's a cache and some problems associated), it's just simple syntax sugar for attr() to avoid writing `.attr('data-test')` everytime and being able to simply write `.data('test')`. The footprint is the same as `attr()`

- Clone the repository
- Create the corresponding folder in `src/plugins`
- Execute `grunt watch` in the terminal. Make sure you have [phantomjs](http://phantomjs.org/) installed first.
- Create the three files, `data.js`, `documentation.md` and `test.js`
- In `data.js` add the code needed to execute it
- Then (or before if you are into TDD) add the needed tests in `test.js`. After adding them all, open tests.html in your browser, which should show a passing/failing for your tests. Or check the terminal, since grunt should run tests now too.
- Add the `documentation.md` needed to use it
- Send a Pull Request and I'll provide feedback on the code

Note: Check out `.attr()` code to create `.data()`

### PR DESCRIPTION
**Pull request by tr3intaydos**
New feature, fixes #4.

### CODE REVIEW

File: src/plugins/data/data.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Retrieve or set the data-* attributes of the first matched node
+ * @param String name the data-* attribute to search
+ * @param String value optional atribute to set
+ * @return this|String
+ */
+// ATTR
+// Return the fist node attribute
+u.prototype.data = function(name, value) {
+  if (typeof name === 'object') {
+    new_name = {};
```
Comment (franciscop) [inline src/plugins/data/data.js]: `var` is missing here

File: src/plugins/data/data.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    for(var key in name) {
+      new_key = 'data-' + key;
+      new_name[new_key] = name[key];
```
Comment (franciscop) [inline src/plugins/data/data.js]: these two lines could be combined, avoiding having to create a new variable like this:

```js
new_name['data-' + key] = name[key];
```

**Comment by franciscop** [comment]
This is looking really good, a couple of notes:

- There are some leftovers in the test (beforeEach, afterEach, typeof base.after, etc)
- See the two notes in the code

Everything else looks perfect, I'll merge it when changed, thank you for the pull request!

**Comment by tr3intaydos** [comment]
Should i fix it two in the .attr() plugin?

**Comment by franciscop** [comment]
Oh wow, totally, I completely missed that! Thanks :smile:

**Comment by franciscop** [comment]
perfect :+1:

**Pull request merged**